### ISSUE TITLE
data(colors) is a poorly chosen name

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1403]
Sheesh. `colors()` is a built-in (and important) R function, and we ought not to be hijacking that name in oce. I seriously doubt that users will need the dataset, except indirectly through e.g. `oceColorsOxygen()` and similar, so I think a name change will be ok. (Do codevelopers agree? If so, please thumbs-up or thumbs-down.)

PS. the help page for `colors` mentions `oceColorsViridis` but no other function. That sentence needs an "e.g." in it; we don't need to list them all because there is a family of those functions, so just giving one link lets users get an overview of them all.

```R
> library(oce)
> data(colors)
> str(colors,1)
List of 13
 $ viridis    : chr [1:256] "#440154" "#440256" "#450457" "#450559" ...
 $ cdom       : chr [1:256] "#F8EEC0" "#F8EDBF" "#F8EBBD" "#F7EABC" ...
 $ chlorophyll: chr [1:256] "#DBF7D8" "#DAF6D6" "#D9F5D5" "#D7F4D3" ...
 $ density    : chr [1:256] "#E5F1F1" "#E3F0F0" "#E2EFF0" "#E0EEEF" ...
 $ freesurface: chr [1:256] "#0E0C71" "#100D75" "#110E79" "#130F7C" ...
 $ oxygen     : chr [1:256] "#410101" "#430201" "#450201" "#460201" ...
 $ par        : chr [1:256] "#321417" "#331517" "#341518" "#361518" ...
 $ phase      : chr [1:256] "#BA9112" "#BC9011" "#BE8F10" "#C08D0F" ...
 $ salinity   : chr [1:256] "#2B1470" "#2C1472" "#2C1574" "#2C1576" ...
 $ temperature: chr [1:256] "#0B222E" "#0B2330" "#0B2432" "#0A2533" ...
 $ turbidity  : chr [1:256] "#EFF4A0" "#EEF39F" "#EEF29D" "#EDF09C" ...
 $ velocity   : chr [1:256] "#132037" "#152139" "#16233C" "#17243F" ...
 $ vorticity  : chr [1:256] "#141F3A" "#14213B" "#15233C" "#16253E" ...
```

**Comment by dankelley** [comment]
Also, the source data have changed, evidently a few months after incorporation into `oce`.

Here is the README in the directory where we import the data into oce:

```
Emit:create_data/colors[dk]$ cat cmocean/README.md
These files were downloaded 2015-09-29 by Dan Kelley from the ``cmocean/rgb``
directory of the [cmocean](https://github.com/kthyng/cmocean) python
application, and are used in oce (to be distributed under GPL>=2) according to
permission granted by the ``cmocean`` author, Kristen Thyng, as stated in an
email sent to Dan Kelley and Clark Richards September 29, 2015.
```

and here is the start of the data as downloaded 2015-09-29:
```
Emit:create_data/colors[dk]$ head cmocean/Oxygen-rgb.txt 
2.561123127838653146e-01 5.154557132735668526e-03 2.220406012176586851e-03
2.625615869631610644e-01 6.229841197612821790e-03 2.588611295074054811e-03
2.689919109436942901e-01 7.351462080372067089e-03 2.974626604640448566e-03
2.754187562110408383e-01 8.491113022252591916e-03 3.367374167428128021e-03
2.818345757864350598e-01 9.664140554494875246e-03 3.772597167520155229e-03
2.882430132354636965e-01 1.086424164475233850e-02 4.187675141802257514e-03
2.946498316791332783e-01 1.207988009421660618e-02 4.607842778118506274e-03
3.010451270282203939e-01 1.333282772998858716e-02 5.041634172935165019e-03
3.074457007649086249e-01 1.458689331351215208e-02 5.474379616775534863e-03
3.138396208422337130e-01 1.586867042508604414e-02 5.916628942043782556e-03
```

But here is what I get today:
```R
> head(read.table("https://raw.githubusercontent.com/matplotlib/cmocean/master/cmocean/rgb/oxy-rgb.txt",header=FALSE))
         V1         V2         V3
1 0.2503218 0.02046237 0.01966892
2 0.2555648 0.02111591 0.02090008
3 0.2608016 0.02176437 0.02215967
4 0.2660446 0.02238304 0.02344016
5 0.2765326 0.02356134 0.02606762
6 0.2817862 0.02410204 0.02740685
```

My feeling (again, I'd like votes by co-developers) is that we ought to try to keep oce in sync with the matplotlib colour scheme. The data are not large, though, and if co-developers want, then I could code it so that e.g.
```R
oceColorsViridis(n)
```
would get an extra argument, perhaps named `version`, to select the 2015 version or some other one. That way, people could get results as before (e.g. to match a publication).  There would be some question as to the best default ... I'd argue that the default should be the up-to-date one.  That would mean that old code might need to be rewritten, but the docs could state a recommendation that people use `version` in their code, to prevent problems in the future.

**Comment by richardsc** [comment]
(In hindsight we should have just made a new package for *only* the `cmocean` colors ...)

I don't really care about whether the actual *data* are there. I didn't even know, TBH. You can always get it by doing e.g. `oceColorsViridis(256)`.

I don't want to get into the business of maintaining old versions of the colors (again, this would be better suited to an add-on package). Maybe we document when they get updated, and leave it to the user to determine if their old code needs to be tweaked (and if they have to tweak it for `version=`, it's not much different to just find the older version, or install an older version of `oce`, etc).

**Comment by dankelley** [comment]
I am coming around to just sticking with the old colours, and saying in the docs how to get a new scheme with a `read.table()` to a URL, etc.; that way, people can make up a scheme themselves, which is another good thing. Kind of "let them eat frozen fish if they want, but also teach them to fish so they can make sushi and get worms".

I just really don't like the idea of someone submitting a paper, then redoing a figure at the revision stage and finding that things change.  I've not looked into these colours in any detail.

**Comment by dankelley** [comment]
Fixed in "dk" branch (commit be0a06b66af26af6d46d6eb7f3524f3cc0a21fcf); the data are now called `ocecolors`, which doesn't make a conflict with the built-in `colors()` function.

I exposed the generator function (or "function closure") named `oceColorsClosure` that is used to create e.g. `oceColorsTemperature()`. I renamed it and also made it more general. Plus, its documentation demonstrates how to download the latest `cmcolor` rgb coefficients and create functions that can be used in e.g. `imagpe()`, just like how e.g. `oceColorsTemperature` et al. can be used.

After some thought, I decided *not* to alter the default colours, because I don't want to get into a situation where `oce` had to be updated whenever `cmcolors` gets updated, and I also don't want users to find that updating `oce` changes the colours in their diagrams.

As is my working mode nowadays, I plan to leave this issue open until "dk" gets merged into "develop", likely before noon on the upcoming Saturday.

PS. I changed "colours" to "colors" throughout the documentation, because that's how the functions are named (in oce and, mostly, in R).
